source/add-existing.txt

```diff
@@ -0,0 +1,232 @@
+
+.. code-block:: sh
+   
+   bundle install
+
+{+odm+} Configuration
+~~~~~~~~~~~~~~~~~~~~~
+
+Generate the default {+odm+} configuration by running the following
+command:
+
+.. code-block:: sh
+   
+   bin/rails g mongoid:config
+
+This generator creates the ``config/mongoid.yml`` configuration file
+used to configure the connection to the MongoDB deployment and the
+``config/initializers/mongoid.rb`` initializer file that you can use
+to set other options.
+
+In the ``config/mongoid.yml`` file, specify your connection string and
+other connection options.
+
+Modify Frameworks
+~~~~~~~~~~~~~~~~~
+
+Open the ``config/application.rb`` file and examine the contents. If it
+requires all the components of Rails by including the ``require
+'rails/all'`` specification, change this to specify individual
+frameworks.
+
+To verify the contents of ``rails/all`` for Rails 7, see the
+:github:`rails GitHub repository
+</rails/rails/blob/7-0-stable/railties/lib/rails/all.rb>`.
+
+The following code is a sample ``config/application.rb`` file that
+demonstrates how to specify individual frameworks instead of using ``rails/all``:
+
+.. code-block:: ruby
+   :caption: config/application.rb
+
+   # Remove or comment out rails/all
+   #require "rails/all"
+ 
+   # Add the following instead of rails/all:
+   require "rails"
+ 
+   # Comment out unneeded frameworks
+   # require "active_record/railtie" rescue LoadError
+   # require "active_storage/engine" rescue LoadError
+   require "action_controller/railtie" rescue LoadError
+   require "action_view/railtie" rescue LoadError
+   require "action_mailer/railtie" rescue LoadError
+   require "active_job/railtie" rescue LoadError
+   require "action_cable/engine" rescue LoadError
+   # require "action_mailbox/engine" rescue LoadError
+   # require "action_text/engine" rescue LoadError
+   require "rails/test_unit/railtie" rescue LoadError
+
+.. note::
+
+   Because they rely on ActiveRecord, the `ActionText
+   <https://guides.rubyonrails.org/action_text_overview.html>`__,
+   `ActiveStorage <https://edgeguides.rubyonrails.org/active_storage_overview.html>`__ and
+   `ActionMailbox
+   <https://guides.rubyonrails.org/action_mailbox_basics.html>`__
+   adapters cannot be used alongside {+odm+}.
+
+Disable ActiveRecord
+~~~~~~~~~~~~~~~~~~~~
+
+Review your application's configuration files, such as
+``config/application.rb``, then remove or comment out any references to
+``config.active_record`` and ``config.active_storage``.
+
+Adjust Models
+~~~~~~~~~~~~~
+
+If your application already uses models, you must adjust them to
+migrate from using ActiveRecord to {+odm+}.
+
+ActiveRecord models derive from ``ApplicationRecord`` and do not have
+column definitions. {+odm+} models generally have no superclass but must
+include the ``Mongoid::Document`` attribute. {+odm+} models usually
+define fields explicitly. You can also use :ref:`dynamic fields
+<mongoid-dynamic-fields>` instead of explicit field definitions.
```

Comment:
S: is there a way to tie these four sentences together more? i'm guessing it's explaining how you need to change your models to use Mongoid, but it reads more like a list--in which case, maybe just make it an actual list?

Reply:
will workshop